Source/PluginProcessor.cpp

```diff
@@ -226,7 +311,7 @@ SynthesiserSound::Ptr SoomplerAudioProcessor::getSampleData(std::optional<File>
 
     BigInteger midiNotes;
     midiNotes.setRange(0, 126, true);
-    return new SamplerSound(sampleFile->getFileName(), *formatReader, midiNotes, 0x40, 0.0, 0.0, 10.0);
+    return new soompler::ExtendedSound(sampleFile->getFileName(), *formatReader, midiNotes, 0x40, 0.0, 0.0, 400.0);

```

Comment:
Who should delete memory? 
No raw pointers

Reply:
This object returned as ExtendedSound::Ptr, and being managed by synth object. Heres what comment of synth.setSound function says: 
> 
>         The object passed in is reference counted, so will be deleted when the
>         synthesiser and all voices are no longer using it.